### PR TITLE
feat: improve error and feedback when target does not exist

### PR DESCRIPTION
**Pull request by tdejager**
Fixes issue: https://github.com/prefix-dev/pixi/issues/1928

Even though what was happening there was intended behavior, there was an expect and just general unclearness. Although I feel the UX could be improved further, this improves the feedback at least. Now giving you the error:

```
pixd remove flask --platform linux-64                                                                                                                                                                                                                                                                  ─╯
  × failed to remove dependency
  ╰─▶ No target for feature `default` found on platform `linux-64`
  help: Expected target for `default`, e.g.: `[target.linux-64.dependencies]
  ```

Given this toml:

```
[project]
name = "bug"
channels = ["conda-forge"]
platforms = ["linux-64"]

[dependencies]
flask = ">=3.0.3,<4"
```

When calling the equivalent command from the issue. This should be enough to improve the situation for now I think.

### CODE REVIEW

File: crates/pixi_manifest/src/manifest.rs [hunk 1]
```diff
@@ -463,22 +471,35 @@ impl Manifest {
         Ok(())
     }
 
+    /// Handles the target missing error cases
     fn handle_target_missing(
         platform: Option<&Platform>,
         feature_name: &FeatureName,
+        section: &str,
```
Comment (nichmor) [inline crates/pixi_manifest/src/manifest.rs:478]: wdyt about making section an enum? I think we have a finite list of supported sections, so maybe having a more strict variant will help with development?

File: src/cli/remove.rs [hunk 2]
```diff
@@ -41,21 +42,27 @@ pub async fn execute(args: Args) -> miette::Result<()> {
...
-                )?;
+                project
+                    .manifest
+                    .remove_dependency(
+                        name,
+                        spec_type,
+                        &dependency_config.platform,
+                        &dependency_config.feature_name(),
+                    )
+                    .context("failed to remove dependency")?;
```
Comment (nichmor) [inline src/cli/remove.rs:65]: maybe?

`.context(format!("failed to remove dependency {name}")) `

File: src/cli/remove.rs [hunk 3]
```diff
@@ -41,21 +42,27 @@ pub async fn execute(args: Args) -> miette::Result<()> {
...
-                    &dependency_config.feature_name(),
-                )?;
+                project
+                    .manifest
+                    .remove_pypi_dependency(
+                        name,
+                        &dependency_config.platform,
+                        &dependency_config.feature_name(),
+                    )
+                    .context("failed to remove pypi dependency")?;
```
Comment (nichmor) [inline src/cli/remove.rs:52]: maybe?

`.context(format!("failed to remove pypi dependency {name}"))`

**Review by nichmor — APPROVED** [review]
super nice!

File: src/cli/remove.rs (hunk 2 above)
Comment (tdejager) [reply src/cli/remove.rs]: Good idea!

File: src/cli/remove.rs (hunk 3 above)
Comment (tdejager) [reply src/cli/remove.rs]: Good Idea!

File: crates/pixi_manifest/src/manifest.rs (hunk 1 above)
Comment (tdejager) [reply crates/pixi_manifest/src/manifest.rs:482]: I changed it to use constants instead, added `dependencies` constant should add that to more places in a later PR.

**Pull request merged**